Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h

```diff
@@ -290,6 +289,8 @@ class UnifiedPDFPlugin final : public PDFPluginBase, public WebCore::GraphicsLay
     };
     SelectionTrackingData m_selectionTrackingData;
     RetainPtr<PDFSelection> m_currentSelection;
+
+    bool m_windowIsActive { true };
```

Comment:
Can we just fetch this on demand, rather than aliasing the state here (which is bug-prune). Does the Page know?

Reply:
Yes, I think we totally can. I see that `WebPage::activityState()` is a thing, so I'll just consult that when we need.

Reply:
Instead of storing state, I just have the WebPage notify the plugin that window activity has changed, so that we can issue a repaint request.

Also, we refer to the activity state information on the `Page` object to determine whether to draw active or inactive colors.